snap-core/src/main/java/org/esa/snap/core/util/ImageUtils.java

```diff
@@ -249,19 +249,26 @@ public static Dimension computeSceneRasterSize(int defaultSceneRasterWidth, int
     }
 
     public static int computeTileCount(int imageSize, int tileSize) {
-        int tileCount = imageSize / tileSize;
+        /*int tileCount = imageSize / tileSize;
         if (imageSize % tileSize != 0) {
             tileCount++;
         }
-        return tileCount;
+        return tileCount;*/
+        return Math.round(imageSize / (float) tileSize);
     }
 
     public static double computeLevelSizeAsDouble(int sourceSize, int level) {
-        return sourceSize / Math.pow(2, level);
+        // return sourceSize / Math.pow(2, level);
```

Comment:
here the commented code could remain, because it is commented why it way replaced and as a comparison it is helpful.

But in the only the actual code matters. So, it can be removed too.

But a unit level test would ensure that the different implementation does not change the behavior.

Reply:
Updates made, unit test added.